src/System.Private.CoreLib/shared/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -463,6 +463,27 @@ private void HandleSatelliteMissing()
             throw new MissingSatelliteAssemblyException(SR.Format(SR.MissingSatelliteAssembly_Culture_Name, _mediator.NeutralResourcesCulture, satAssemName), missingCultureName);
         }
 
+        private static string GetManifestResourceNamesList(Assembly assembly)
+        {
+            try
+            {
+                string postfix = "\"";
+                string []? resourceSetNames = assembly.GetManifestResourceNames();
+
+                // If we have more than 10 resource sets, we just print the first 10 for the sake of the exception message readability.
+                if (resourceSetNames.Length > 10)
```

Comment:
Do you need a null check here since `resourceSetNames` can be null according to the annotation?

Reply:
Actually, the annotation on `GetManifestResourceNames` says that it won't return null. It means that the type of the `resourceSetNames` should be without question mark.

Reply:
Also you can remove the exclamation mark below.

Reply:
I don't think I need this check. GetManifestResourceNames always return an array and never return null. I just made this type nullable only to make the compiler happy when I pass resourceSetNames  to the other APIs (String.Join and Array,Resize)

Reply:
The nullable annotations should express the intent. If the variable is non-null, it should be marked as non-null. The exclamation marks should be used for uses where the compiler is not happy.

None of this should be an issue once if you switch to Slice instead Array.Resize.

Reply:
I did @jkotas suggestion.